lib/src/rules/library_private_types_in_public_api.dart

```diff
@@ -209,6 +209,27 @@ class Validator extends SimpleAstVisitor<void> {
     node.type?.accept(this);
   }
 
+  @override
+  void visitSuperFormalParameter(SuperFormalParameter node) {
+    if (node.isNamed && Identifier.isPrivateName(node.identifier.name)) {
+      return;
+    }
+
+    // Check for a declared type.
+    var type = node.type;
+    if (type != null) {
+      type.accept(this);
+      return;
+    }
+
+    // Check implicit type.
```

Comment:
Do we also need to check the implicit type in a `FieldFormalParameter`? For example:

```dart
class C {
  _O _x;

  C(this._x);
```

Reply:
Funny, I thought about that and then convinced myself that the declaration would be checked but that's not right, so yes!